openhands/core/config/sandbox_config.py

```diff
@@ -31,6 +32,7 @@ class SandboxConfig:
     """
 
     api_hostname: str = 'localhost'
```

Comment:
If we are adding this, maybe we should deprecate the `api_hostname`?

Reply:
Basically, we can rename `remote_runtime_url` to `runtime_url` and default it to localhost, and change the following to `self.runtime_url`:
https://github.com/All-Hands-AI/OpenHands/blob/5ccee7c8a7d8eff9f8109d8968cf4b316433bc61/openhands/runtime/client/runtime.py#L129-L131

Reply:
So there are two URLs that are important here:
* The URL of the remote runtime API
* The URL of the EventStream API inside of a remote runtime

AFAICT we were kind of reusing the same options for both, but should probably separate them into two options. WDYT?

Reply:
The original idea of `api_hostname` was trying to mainly support `RemoteRuntime` -- if we want to explicitly have a  `remote_runtime_url`, then I think maybe it'll be good to get rid of `api_hostname`, merge it together, or rename it into something clearer?. Otherwise having two URL/Hostname in the same SandboxConfig could be a little confusing

I think when running inside `app` container, the `api_hostname` will be changed to `cluster.host` or something similar that points to the host machine..